### DISCUSSION TITLE
Tag below Tab

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#34] [Ideas]
If a tab has a tag applied show the tag below the tab. Maybe the same height as the tab?

**Comment by madprops** [comment]
Seems a bit weird to double the height of an item, especially when it might just be 1 short tag.
How about add an icon to tagged tabs, and when hovering the tag icon you get all tag information?

**Comment by N3C2L** [comment]
Yeah I thought about that too, but thing is, I want to use the tags to quickly add to-do-thing tags to the tabs. And when I am scrolling through my tabs I could have a good overview of this to-dos if a separate line for tags exist. This separate tag line could be powered with options like: left click: edit this tag, right click: spawn the recent tag list, middle click: delete the tag.

**Comment by madprops** [comment]
Ok this is in progress. Wouldn't left click be better for filtering tabs with this tag instead of edit?
Or maybe left click shows a menu with various options.

**Reply by N3C2L** [reply]
I would love to edit the tag directly when left click. But additionally maybe something similar like the hover button function on the tabs could be implemented on the left side of the tag line. For menu things or filtering tabs.

**Comment by madprops** [comment]
I think i'll keep the menu. Editing a single tag is not supported since you edit the whole tag string directly, and it's easy.

![taglist2w34](https://github.com/madprops/grasshopper/assets/206498/e41f7705-933b-4fc4-abd3-58e4fcdb2e54)

Also filter would be much more useful to me than edit.

Maybe I'll add an edit mode.

**Comment by N3C2L** [comment]
Thanks. I think I misunderstood the tags. I thought a single tag exist. So if you remove a tag from a single tab then this tag is also removed from other tabs? Right now a remove command for a tag does not exist, only remove all tags together.

**Reply by madprops** [reply]
Each tab can have one or more tags. To remove a tag from a tab you edit the tags and manually remove it from the list. Removing a tag from a tab doesn't affect the other tabs.

I added `Taglist Mode` which has menu, filter, edit, remove, modes.

Also editing a tag in a tab doesn't affect the other tags.

Maybe i'll add a command to remove a specific tag from all tabs.

**Comment by N3C2L** [comment]
Thanks for the great explanations, looking forward to test the tags.

**Comment by madprops** [comment]
It's ready. I like how it turned out, it was a good idea.

**Comment by N3C2L** [comment]
Very good. I like it. 
One thing: If the hover button is set to left, it overlaps the website titles and the tags titles a little bit. I would be grateful if you could reduce the size of the hover button a little bit.

**Comment by madprops** [comment]
Hover Button is now smaller.

**Comment by N3C2L** [comment]
Thanks a lot!
